### PR TITLE
[STORM-1005] set storm.local.dir relative values relative to storm.home rather tha…

### PR DESCRIPTION
**Pull request by Renkai**
Solve the same problem with 
https://github.com/apache/storm/pull/713
but another way as @HeartSaVioR advised.

### CODE REVIEW

**Comment by HeartSaVioR** [comment]
@Renkai 
Great. Could you add simple unit test to check ```absolute-storm-local-dir``` works with relative / absolute path?

**Comment by Renkai** [comment]
@HeartSaVioR I hope the unit test is good enough,but I don't have a windows environment to test the code.It seems travis-ci do not have windows environment,too.

**Comment by HeartSaVioR** [comment]
+1
@Renkai I think current unit test is enough.

**Comment by Renkai** [comment]
Would you like to merge these commits or they need someone else to review?

**Comment by HeartSaVioR** [comment]
@Renkai 
Sorry for the delay. I intended to wait for other committers to review, but since no one is reviewing I'll take care of it.

File: storm-core/test/clj/backtype/storm/config_test.clj [hunk 1]
```diff
@@ -168,3 +168,24 @@
...
+(deftest test-absolute-storm-local-dir
+  (let [storm-home-key "storm.home"
+        old-storm-home (System/getProperty storm-home-key)
+        conf-relative {STORM-LOCAL-DIR "storm-local"}
+        conf-absolute {STORM-LOCAL-DIR
+                       (if on-windows?
+                         "C:\\storm-local"
+                         "/var/storm-local")}]
+    (try
+      (System/setProperty storm-home-key (if on-windows? "C:\\storm-home" "/usr/local/storm-home"))
```
Comment (HeartSaVioR) [inline storm-core/test/clj/backtype/storm/config_test.clj]: @Renkai 
Unfortunately I found this PR affects other tests randomly - supervisor tests are failing. 
Please see https://travis-ci.org/apache/storm/jobs/79231189.
Could we just use current storm.home?

Comment (HeartSaVioR) [inline storm-core/test/clj/backtype/storm/config_test.clj]: @Renkai If test should change storm.home to work correctly, let's remove test.

**Comment by Renkai** [comment]
@HeartSaVioR It's fine to leave default `storm.home` to test,I manually set it before because I'm not sure if the value is already set when the test code was running.Now I remove the code which may cause environment changing.

**Pull request merged**